From b549383fab7131c4a8a4a3b3f268ab42b06d577a Mon Sep 17 00:00:00 2001
From: Honggang Li <honli@redhat.com>
Date: Sun, 10 Jul 2016 23:39:38 -0400
Subject: [PATCH] opafmd: larger array to hold program path

Since we're substituting /opt -> /usr/lib, char[25] is no longer
sufficient to hold the complete path.

Signed-off-by: Honggang Li <honli@redhat.com>
---
 Esm/ib/src/linux/startup/opafmd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Esm/ib/src/linux/startup/opafmd.c b/Esm/ib/src/linux/startup/opafmd.c
index 190de4a..945560b 100644
--- a/Esm/ib/src/linux/startup/opafmd.c
+++ b/Esm/ib/src/linux/startup/opafmd.c
@@ -424,7 +424,7 @@ int parseInput(char *buf){
  */
 int spawn(const unsigned int instance, const int component, int *pids){
 	int pid;
-	char prog[25], name[6];
+	char prog[50], name[6];
 	if (instance >= 4){
 		fprintf(stderr, "Invalid instance number.\n");
 		return -1;
-- 
2.7.4

